src/twister2/builder/build_manager.py

```diff
@@ -110,6 +115,11 @@ def build(self) -> None:
         if status == BuildStatus.DONE:
             logger.info('Already build in %s', self.build_config.build_dir)
             return
+        if status == BuildStatus.SKIPPED:
```

Comment:
When two tests use the same source (same build directory), and first test has been skipped due to `TwisterMemoryOverflowException` then the second test will loose an information why it is skipped. We should consider how to keep such information in the build report. 
Maybe we should have predefined all skip reasons in some dictionary and keep an reference in the report, so that we could raise proper exception.
For example:
```python
class SkipReason(Enum):
    default = 0
    memory_overflow_exception = 1
    ...
```

Reply:
Yes, I also thought about this case. At this moment I don't have better idea for this. Implementation of this system can be quite demanding and I don't know what is our priorities - this distinguish of skip reasons concerns only pure pytest tests and as I understood we should focus mostly on yaml tests where this problem does not occur.

I think that this is more complicated problem, because similar case is connected with failed status. When two tests base on the same build and it failed, then only one test will print full logs about what exactly failed during building. The second one will print only short information that `Found in {self._status_file} the build status is set as {BuildStatus.FAILED} 
 for: {self.build_config.build_dir}`. And this information is also not too verbose - users even don't know where they should look for potential information about more details.
 
And I completely agree, that we should save more details about what was the reason of skip or fail, but I think that this can be done in other task/PR.